utils/src/server/rpc_module.rs

```diff
@@ -397,48 +391,37 @@ impl Methods {
 	///         sink.send(&"one answer").unwrap();
 	///         Ok(())
 	///     }).unwrap();
-	///     let (resp, mut stream) = module.call_and_subscribe("hi", EmptyParams::new()).await.unwrap();
+	///     let (resp, mut stream, _) = module.raw_call("hi", EmptyParams::new().to_rpc_params().unwrap()).await.unwrap();
 	///     assert!(serde_json::from_str::<Response<u64>>(&resp).is_ok());
 	///     let raw_sub_resp = stream.next().await.unwrap();
 	///     let sub_resp: SubscriptionResponse<String> = serde_json::from_str(&raw_sub_resp).unwrap();
 	///     assert_eq!(&sub_resp.params.result, "one answer");
 	/// }
 	/// ```
-	pub async fn call_and_subscribe<Params: ToRpcParams>(
-		&self,
-		method: &str,
-		params: Params,
-	) -> Option<(String, mpsc::UnboundedReceiver<String>)> {
+	pub async fn raw_call(&self, method: &str, params: Box<RawValue>) -> RawRpcResponse {
```

Comment:
this is still confusing I think...

Reply:
How about this reasoning: instead of struggling to find a good name for these, how about we give them crystal clear but ugly names that clearly signals what they are: building blocks for tests. So in the vein of `test_subscription`, we can call this `test_method_call` or `method_call_helper`.

Reply:
hehe, this could be either of those :)

if you look at the code now, both `call` and `test_subscription` uses `raw_call` under the hood thus more or less similar to `handle_request` in the `jsonrpc` crate.

however, the best is if we ditch this in the public API and just have `call` and `subscribe` I guess but won't to send raw json.

James has a point that these might be useful else than only in tests, https://github.com/paritytech/jsonrpsee/pull/458#discussion_r707126871

Reply:
How about:
`test_subscription` –> `subscription_helper`
`raw_call` –> `method_call_helper`

@maciejhirsz wdyt?

Reply:
can you read the PR description and skim the code again?

I made some changes and updated the PR description right now.